pages/api/shoppinglist/[id]/index.js

```diff
@@ -14,5 +14,11 @@ export default async function handler(request, response) {
     response.status(200).json(product);
     return;
   }
+  if (request.method === "DELETE") {
+    const result = await shoppingItem.findByIdAndDelete(id);
```

Comment:
you are not using your variable result here, you can remove it and just write `await shoppingItem.findById...`